bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -85,6 +85,31 @@ def run_plan(scope, plan_name, named_args = {})
       executor.report_yaml_plan(closure.model.body)
     end
 
+    # If a TargetSpec parameter is passed, ensure it is in inventory
+    inventory = Puppet.lookup(:bolt_inventory)
+    # Is it possible that inventory is not here?
```

Comment:
Given we don't handle it being nil in any other function, I don't think so. Inventory is created when the CLI starts (unless it's just `--help`), and then loaded using `with_bolt_executor` in `Bolt::Pal`, which all plans use as part of executing the plan (via `in_plan_compiler`). I don't see any way this doesn't get set, at least currently.